include/odp/api/spec/threshold.h

```diff
@@ -0,0 +1,105 @@
+/* Copyright (c) 2017, Linaro Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+/**
+ * @file
+ *
+ * ODP threshold descriptor
+ */
+
+#ifndef ODP_API_THRESHOLD_H_
+#define ODP_API_THRESHOLD_H_
+#include <odp/visibility_begin.h>
+#include <odp/api/std_types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Supported threshold types
+ *
+ * Supported threshold types in a bit field structure.
+ */
+typedef union odp_threshold_types_t {
+	/** bitfields for different threshold types */
+	struct {
+		/** Percentage of the total size of pool or queue */
+		uint8_t percent:1;
+
+		/** Total number of all transient packets */
+		uint8_t packet:1;
+
+		/** Total size of all transient packets in bytes */
+		uint8_t bytes:1;
+	};
+
+	/** All bits of the bit field structure */
+	uint8_t all_bits;
+} odp_threshold_types_t;
+
+/**
+ * ODP Threshold types
+ *
+ * Different types of threshold measurements
+ */
+typedef	enum {
+	/** Percentage of the total size of pool or queue */
```

Comment:
Change <tab> to <space> between typedef and enum. Also typically the typedef'ed enum/struct/union has also name, so that a debugger does not refer to it as "enum <no name> does not ...".

typedef enum odp_threshold_type_t {
  ...
} odp_threshold_type_t;